accumulator_updater/tests/accumulator_updater.ts

```diff
@@ -180,16 +228,54 @@ function parseAccumulatorInput({
 }: {
   header: AccumulatorInputHeader;
   data: Buffer;
-}): AccumulatorInputPriceAccountTypes {
+}): AccumulatorPriceAccount {
   console.log(`header: ${JSON.stringify(header)}`);
   assert.strictEqual(header.accountType, 3);
   if (header.accountSchema === 0) {
     console.log(`[full]data: ${data.toString("hex")}`);
-    // case-sensitive. Note that "P" is capitalized here and not in
-    // the AccumulatorInputPriceAccountTypes type alias.
-    return mockCpiProg.coder.accounts.decode("PriceAccount", data);
+    return parsePriceFull(data);
   } else {
     console.log(`[compact]data: ${data.toString("hex")}`);
-    return mockCpiProg.coder.types.decode("PriceOnly", data);
+    return parsePriceCompact(data);
   }
 }
+
+//TODO: follow wormhole sdk parsing structure?
+// - https://github.com/wormhole-foundation/wormhole/blob/main/sdk/js/src/vaa/generic.ts
+type AccumulatorPriceAccount = PriceFull | PriceCompact;
```

Comment:
fwiw i find this name confusing. shouldn't this be something very generic like `Message`? the accumulator basically stores a bunch of arbitrary messages (each with their own schema)?

Reply:
actually i think this comment applies more generally, as the names currently make it unclear what things are generic and what things are specific to price accounts. I think the code would be easier to follow if eg. `PythSchema` was named something generic like `MessageSchema`, and then the stuff in `price.rs` was `message.rs`. The point here is to be clear via the naming and code structure what is generic data in the accumulator vs. what is pyth specific.

Reply:
I agree, the naming should definitely be cleared up. There is the generic `AccumulatorInput` account that is 
```
AccumulatorInput: {
   header: AccumulatorHeader,
   data: Vec<u8> 
}
```

There is a little bit a blurry area though since I'm assuming we may want to be flexible and support being able to add `Product/Mapping Accounts` into the accumulator. Even though everything going into the `AccumulatorInput` PDAs are generic "messages", we do need to eventually map the data in these back into concrete `(PythAccountType, PythSchema)` structs ("PriceFull", "ProductCompact", etc).